src/controllers/sessionController.ts

```diff
@@ -3,38 +3,24 @@
  */
 
 import * as express from "express";
-import { isLeft } from "fp-ts/lib/Either";
 import {
-  IResponseErrorInternal,
-  IResponseErrorNotFound,
+  IResponseErrorValidation,
   IResponseSuccessJson,
-  ResponseErrorInternal,
   ResponseSuccessJson
 } from "italia-ts-commons/lib/responses";
-import { PublicSession } from "../types/api/PublicSession";
-import { extractUserFromRequest } from "../types/user";
 
-export default class SessionController {
-  public async getSessionState(
-    req: express.Request
-  ): Promise<
-    | IResponseErrorInternal
-    | IResponseErrorNotFound
-    | IResponseSuccessJson<PublicSession>
-  > {
-    const errorOrUser = extractUserFromRequest(req);
-
-    if (isLeft(errorOrUser)) {
-      const error = errorOrUser.value;
-      return ResponseErrorInternal(error.message);
-    }
+import { PublicSession } from "../../generated/backend/PublicSession";
 
-    const user = errorOrUser.value;
+import { withUserFromRequest } from "../types/user";
 
-    // Return the actual session information.
-    return ResponseSuccessJson({
-      spidLevel: user.spid_level,
-      walletToken: user.wallet_token
-    });
-  }
+export default class SessionController {
+  public readonly getSessionState = (
+    req: express.Request
+  ): Promise<IResponseErrorValidation | IResponseSuccessJson<PublicSession>> =>
+    withUserFromRequest(req, async user =>
```

Comment:
nit: if you curry `withUserFromRequest(callback)(request)` you can use directly
`getSessionState = withUserFromRequest(callback)`

Reply:
care to send a PR with this improvement? 👍